Modules/Module_Processing/Module_Population/xASL_stat_GetROIstatistics.m

```
@@ -277,7 +277,9 @@
             TotalRows = x.dataset.nSubjects;
 		else
 			x.S.SubjectSessionID{SubjSess,1} = [x.SUBJECTS{iSubject} '_' listSessions{iSess}];
-            TotalRows = x.dataset.nSubjects * x.dataset.nSessions;
+			% The number of sessions have to be consistent with the detected one. If we have 2 sessions, but only session 2 has quantified CBF, 

```

I'm not too sure about this. We rather have "dirty" stats files that show the error rather than cleaning it such that it looks better but the user doesn't note the error.

E.g., if a study has only rawdata run-2, then this should be converted to only having ASL_2 and x.dataset.nSessions should be 1, where x.dataset.SESSIONS = {'ASL_2'};

E.g., if a study has rawdata run-1 and run-2, then this should be converted to ASL_1 and ASL_2, where it would be useful for the user to see for all participants that ASL_1 is n/a.

By the way, I rephrased this because at this point it is not about "printing lines" but it is about which we will attempt to perform stats on, and you rather have a 1000 warnings on missing sessions (which we could perhaps catch and only print once to the screen) rather than catching it here and not processing it.

The alternative is to do this with nSessions (what you propose) but issue a warning that nSessions~=x.dataset.nSessions. We could already issue that warning at the top where we obtain nSessions.

---

If we have ASL_2 only and then obviously only CBF_2, then everything is fine - just a single session is detected and written both for volumes and CBF.

The trouble is when we have ASL_1 and ASL_2. But quantification of ASL_1 fails, so we only have CBF_2. Then the two indexing options do not match and it results in a messed up stats table. Note that the issue is not with run_1 -> ASL_1 conversion. But with quantification of Legacy ASL_1 to CBF_1. 

As discussed -> I only programmed this as a second line safety check. This error should be captured elsewhere and reported. But we can't count that the users will do everything right and check. So if a user ignores errors, then we still have to make sure that errors are not propagated to the final table.

I'm just clarifying the motivation. I agree with the edit you made to the comment and about the extra check of nSession and x.dataset.nSessions -> these are useful 2nd/3rd line checks ;) 

You can resolve after reading.